user-lock-manager-w-keypad-support.smartapp.groovy

```diff
@@ -1066,20 +1092,29 @@ def performActions(evt) {
     def codeData = new JsonSlurper().parseText(evt.data)
     if(enabledUsersArray().contains(codeData.usedCode) || isManualUnlock(codeData)) {
       // Global Hello Home
-      if (noRunPresence && doRunPresence == null) {
-        if (!anyoneHome(noRunPresence)) {
-          location.helloHome.execute(homePhrases)
-        }
-      } else if (doRunPresence && noRunPresence == null) {
-        if (anyoneHome(doRunPresence)) {
-          location.helloHome.execute(homePhrases)
-        }
-      } else if (doRunPresence && noRunPresence) {
-        if (anyoneHome(doRunPresence) && !anyoneHome(noRunPresence)) {
-          location.helloHome.execute(homePhrases)
-        }
-      } else {
-       location.helloHome.execute(homePhrases)
+      if(location.currentMode != homePhrases)
```

Comment:
I apologize for any formatting sloppiness, I unfortunately don't have any code editing program (i.e. Visual Studio, Eclipse, etc.) so this is all edited in the IDE directly. I'll work to clean it up.